lib/router/router.js

```diff
@@ -51,6 +52,12 @@ export default class Router {
   }
 
   async onPopState (e) {
+    // If the downstream application returns falsy, return.
+    // They will then be responsible for handling the event.
+    if (!this._beforePopState(e.state)) {
```

Comment:
Could you move this after the following if condition?

Reply:
Done